pkg/component/istio/monitoring.go

```diff
@@ -80,125 +82,94 @@ const (
 )
 
 var (
-	monitoringAllowedMetricsIstiod = []string{
-		monitoringMetricGalleyValidationFailed,
-		monitoringMetricGalleyValidationPassed,
-		monitoringMetricGoGoroutines,
-		monitoringMetricGoMemstatsAllocBytes,
-		monitoringMetricGoMemstatsHeapAllocBytes,
-		monitoringMetricGoMemstatsHeapInuseBytes,
-		monitoringMetricGoMemstatsHeapSysBytes,
-		monitoringMetricGoMemstatsStackInuseBytes,
-		monitoringMetricIstioBuild,
-		monitoringMetricPilotConflictInboundListener,
-		monitoringMetricPilotConflictOutboundListenerHttpOverCurrentTcp,
-		monitoringMetricPilotConflictOutboundListenerTcpOverCurrentHttp,
-		monitoringMetricPilotConflictOutboundListenerTcpOverCurrentTcp,
-		monitoringMetricPilotK8sCfgEvents,
-		monitoringMetricPilotProxy_convergenceTimeBucket,
-		monitoringMetricPilotServices,
-		monitoringMetricPilotTotalXdsInternalErrors,
-		monitoringMetricPilotTotalXdsRejects,
-		monitoringMetricPilotVirtServices,
-		monitoringMetricPilotXds,
-		monitoringMetricPilotXdsCdsReject,
-		monitoringMetricPilotXdsEdsReject,
-		monitoringMetricPilotXdsLdsReject,
-		monitoringMetricPilotXdsPushContextErrors,
-		monitoringMetricPilotXdsPushes,
-		monitoringMetricPilotXdsRdsReject,
-		monitoringMetricPilotXdsWriteTimeout,
-		monitoringMetricProcessCpuSecondsTotal,
-		monitoringMetricProcessOpenFds,
-		monitoringMetricProcessResidentMemoryBytes,
-		monitoringMetricProcessVirtualMemoryBytes,
+	serviceMonitorIstiod = &monitoringv1.ServiceMonitor{
+		ObjectMeta: monitoringutils.ConfigObjectMeta(monitoringPrometheusJobNameIstiod, v1beta1constants.IstioSystemNamespace, aggregate.Label),
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Selector: metav1.LabelSelector{MatchLabels: getIstiodLabels()},
+			Endpoints: []monitoringv1.Endpoint{{
+				Port: istiodServicePortNameMetrics,
```

Comment:
Is this really doing the same thing as before? If so, that is great as it is much more concise and understandable.

Reply:
As far as I have seen, yes. Why do you think it's not doing the same? :)

Reply:
It is just very much shorter and concise as before. I was wondering if we did something unnecessarily complicated before or if `prometheus-operator` just does the right thing.
Anyway, it is better now.